src/test/java/com/google/cloud/spanner/pgadapter/ITErrorsTest.java

```diff
@@ -0,0 +1,172 @@
+    try (Connection connection = DriverManager.getConnection(getConnectionUrl())) {
+      connection.createStatement().execute("delete from my_other_table");
+      connection.createStatement().execute("delete from my_table");
+    }
+  }
+
+  private String getConnectionUrl() {
+    return String.format(
+        "jdbc:postgresql://%s/%s",
+        testEnv.getPGAdapterHostAndPort(), database.getId().getDatabase());
+  }
+
+  @Test
+  public void testTableNotFound() throws SQLException {
+    try (Connection connection = DriverManager.getConnection(getConnectionUrl())) {
+      PSQLException exception =
+          assertThrows(
+              PSQLException.class,
+              () ->
+                  connection
+                      .createStatement()
+                      .executeUpdate("insert into my_non_existing_table (id, ref) values (1, 1)"));
+      assertNotNull(exception.getServerErrorMessage());
+      assertEquals(
+          SQLState.UndefinedTable.toString(), exception.getServerErrorMessage().getSQLState());
+      assertEquals(SQLState.UndefinedTable.toString(), exception.getSQLState());
+    }
+  }
+
+  @Test
+  public void testColumnNotFound() throws SQLException {
+    try (Connection connection = DriverManager.getConnection(getConnectionUrl())) {
+      PSQLException exception =
+          assertThrows(
+              PSQLException.class,
+              () ->
+                  connection
+                      .createStatement()
+                      .executeUpdate("insert into my_other_table (id, ref) values (1, 1)"));
+      assertNotNull(exception.getServerErrorMessage());
+      assertEquals(
+          SQLState.UndefinedColumn.toString(), exception.getServerErrorMessage().getSQLState());
+      assertEquals(SQLState.UndefinedColumn.toString(), exception.getSQLState());
+    }
+  }
+
+  @Test
+  public void testPrimaryKeyConstraintViolation() throws SQLException {
+    try (Connection connection = DriverManager.getConnection(getConnectionUrl())) {
+      assertEquals(
+          1,
+          connection
+              .createStatement()
+              .executeUpdate("insert into my_table (id, value) values (1, 'One')"));
+      PSQLException exception =
+          assertThrows(
+              PSQLException.class,
+              () ->
+                  connection
+                      .createStatement()
+                      .executeUpdate("insert into my_table (id, value) values (1, 'Other one')"));
+      assertNotNull(exception.getServerErrorMessage());
+      assertEquals(
+          SQLState.UniqueViolation.toString(), exception.getServerErrorMessage().getSQLState());
+      assertEquals(SQLState.UniqueViolation.toString(), exception.getSQLState());
+    }
+  }
+
+  @Test
+  public void testUniqueIndexViolation() throws SQLException {
+    try (Connection connection = DriverManager.getConnection(getConnectionUrl())) {
+      assertEquals(
+          1,
+          connection
```

Comment:
We are copying this assertion across tests. Can we create a separate happy test for this?

Reply:
See below